### ISSUE TITLE
Grafana apt update doesn't have 8.5.16

### ISSUE THREAD

**Issue by filippog** [grafana/grafana#62544]
I was expecting to find 8.5.16 on both the [download page](https://grafana.com/grafana/download/8.5.16?edition=oss) via `dl.grafana.com` and the `apt.grafana.com` repository, the latter isn't updated though:

```
$ curl -s https://apt.grafana.com/dists/stable/main/binary-amd64/Packages.gz | zgrep 'Version: 8\.5\.16'
$
```

**Comment by mellieA** [comment]
@filippog & @hho there were changes to the release repository in Nov 2022 documented here: https://grafana.com/docs/grafana/latest/setup-grafana/installation/debian/#repository-migration-november-8th-2022

I've moved this to our release engineering squad to look into why `apt upgrade` isn't working for the latest releases and if there are extra steps for users that installed using the old apt repository.

As a workaround: 

I was able to get the files with ` wget https://dl.grafana.com/oss/release/grafana_8.5.16_arm64.deb` documented on the [download page](https://grafana.com/grafana/download/8.5.16?edition=oss) for debian users: 

```
sudo apt-get install -y adduser libfontconfig1
wget https://dl.grafana.com/oss/release/grafana_8.5.16_amd64.deb
sudo dpkg -i grafana_8.5.16_amd64.deb
```

**Comment by mellieA** [comment]
A user in https://github.com/grafana/grafana/issues/62595#issuecomment-1410496434 confirmed that the problem exists even when they are using the rotated keys: 

> As a sidenote I'm using the latest rotated gpg keys mentioned here: https://apt.grafana.com/

**Comment by mellieA** [comment]
There is a [blog post](https://grafana.com/blog/2023/01/12/grafana-labs-update-regarding-circleci-security-updates/) to determine if your trust relationship is configured correctly after updating the gpg keys. Can you verify with these steps and report back with the results you see: 
 
> 
> Debian/Ubuntu
> The following command should return ‘0’:
> 
> apt-key list | grep -i grafana | wc -l
> 
> If the above command returned anything except ‘0’, remove the old key:
> 
> # Remove the old key
> apt-key del 4E40DDF6D76E284A4A6780E48C8C34C524098CB6
> 
> # Verify that the GPG key was removed. This should return 0.
> apt-key list | grep -i grafana | wc -l
> Regardless of whether the above command returned ‘0’ or not, you will need to configure the new key as well:
> 
> # Download the new key and place it in the right location
> sudo wget -q -O /usr/share/keyrings/grafana.key https://apt.grafana.com/gpg.key
> 
> # Replace the apt source
> # This command will replace the contents of the grafana.list file.
> echo "deb [signed-by=/usr/share/keyrings/grafana.key] https://apt.grafana.com stable main" | sudo tee /etc/apt/sources.list.d/grafana.list
> 
> # Update your repositories list
> sudo apt update
>

**Comment by filippog** [comment]
Thank you for your help @mellieA in triaging/debugging!

As far as this specific issue goes (i.e. `8.5.16` version is not found at all in the repository) I can confirm that the issue is present on the post-migration repository (i.e. `apt.grafana.com`).

I can't speak to the failure to find 9.3.6 reported in https://github.com/grafana/grafana/issues/62544#issuecomment-1410313442 since that seems a perhaps-related but different type of failure.

I have confirmed we're using the updated gpg keys while fetching updates (even though gpg isn't directly involved/related in the repro steps I mentioned)

To clarify, this is supposed to return some results and it doesn't:

```
$ curl -s https://apt.grafana.com/dists/stable/main/binary-amd64/Packages.gz | zgrep 'Version: 8\.5\.16'
$
```

Whereas `8.5.15` is found as expected:

```
$ curl -s https://apt.grafana.com/dists/stable/main/binary-amd64/Packages.gz | zgrep 'Version: 8\.5\.15'
Version: 8.5.15
Version: 8.5.15
$
```

**Comment by mnul** [comment]
I tried a fresh installation today and used the latest documentation (so no upgrade here). I also get the error
`Err:1 https://apt.grafana.com stable/main amd64 grafana-enterprise amd64 9.3.6
  404  Not Found [IP: 151.101.66.217 443]
E: Failed to fetch https://apt.grafana.com/pool/main/g/grafana-enterprise/grafana-enterprise_9.3.6_amd64.deb  404  Not Found [IP: 151.101.66.217 443]` 

Trying the requested command 
`apt-key list | grep -i grafana | wc -l` returns `0`.

**Comment by mellieA** [comment]
Our release engineering team is actively working on this, stay tuned for updates!

**Comment by mellieA** [comment]
The repo repair was kicked off about two hours ago -  our devs were able to downloaded grafana 9.3.6 and 8.5.15 without issues

For 8.5.16 and 8.5.20 we will republish them manually but we have to wait until the repair is complete. So these should be available tomorrow.

**Comment by paladox** [comment]
> The repo repair was kicked off about two hours ago - our devs were able to downloaded grafana 9.3.6 and 8.5.15 without issues
> 
> For 8.5.16 and 8.5.20 we will republish them manually but we have to wait until the repair is complete. So these should be available tomorrow.

Thanks! It now works for me with 9.3.6!

**Comment by julienduchesne** [comment]
Hello, sorry about all that. Some code changes around GPG signing (due to the new key) caused some issues in our package repositories. `8.5.16` and `8.5.20` should now be available:

## rpm

```
[root@39cbbe9018e8 /]# yum --showduplicates list grafana-enterprise | grep 8.5
...
grafana-enterprise.aarch64                   8.5.16-1                    grafana
grafana-enterprise.armhfp                    8.5.16-1                    grafana
grafana-enterprise.x86_64                    8.5.16-1                    grafana
grafana-enterprise.aarch64                   8.5.20-1                    grafana
grafana-enterprise.armhfp                    8.5.20-1                    grafana
grafana-enterprise.x86_64                    8.5.20-1                    grafana
```

```
[root@39cbbe9018e8 /]# yum --showduplicates list grafana | grep 8.5
...
grafana.aarch64                     8.5.16-1                             grafana
grafana.armhfp                      8.5.16-1                             grafana
grafana.x86_64                      8.5.16-1                             grafana
grafana.aarch64                     8.5.20-1                             grafana
grafana.armhfp                      8.5.20-1                             grafana
grafana.x86_64                      8.5.20-1                             grafana
```

## deb

```
root@3b10717105d9:/# apt list -a grafana-enterprise | grep 8.5
grafana-enterprise/stable 8.5.20 amd64
grafana-enterprise/stable 8.5.16 amd64
...
```

```
root@3b10717105d9:/# apt list -a grafana | grep 8.5
grafana/stable 8.5.20 amd64
grafana/stable 8.5.16 amd64
...
```

**Comment by filippog** [comment]
I can confirm 8.5.20 is available as expected! Thank you @julienduchesne @mellieA ! IMHO we're good to close this issue, I'll let Grafana folks do that as they see fit

**Comment by hho** [comment]
Yes, the workaround was already good and now everything seems fine again.

Huge thanks to @mellieA and @julienduchesne for the super quick turnaround :+1:
